src/components/RequireAuth/RequireAuth.tsx

```diff
@@ -0,0 +1,12 @@
+import useAuth from "../../hooks/useAuth";
+import { Navigate, Outlet, useLocation } from "react-router-dom";
+
+export default function RequireAuth() {
+  const { user } = useAuth();
+  const location = useLocation();
+  return user.accessToken != "" ? (
+    <Outlet />
+  ) : (
+    <Navigate to="/Studyfied/login" state={{ from: location }} replace />

```

Comment:
instead of navigating to the login page, block the current page with a popup asking for a login or account creation